### PR TITLE
fix: Typography `icons` should ignore React.Fragment

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix the problem that Typography.Paragraph copyable will treat react. Fragment as an array |
| 🇨🇳 Chinese | 修复 Typography.Paragraph 的 copyable 会将 React.Fragment 当做数组处理的问题 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
test case~~~

File: components/typography/__tests__/index.copy.test.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
```
Comment (zombieJ) [inline components/typography/__tests__/index.copy.test.js:1]: 文件名直接叫 `copy.test.js` 即可~~

File: components/typography/Base.tsx [hunk 2]
```diff
@@ -391,14 +391,12 @@ class Base extends React.Component<InternalBlockProps, BaseState> {
...
-    if (tooltipNodes.length === 0) {
-      tooltipNodes = [this.copyStr, this.copiedStr];
-    }
+    const { tooltips, icon } = copyable as CopyConfig;
+
+    const tooltipNodes = Array.isArray(tooltips) ? tooltips : [this.copyStr, this.copiedStr];
     const title = copied ? tooltipNodes[1] : tooltipNodes[0];
     const ariaLabel = typeof title === 'string' ? title : '';
-    const icons = toArray((copyable as CopyConfig).icon);
+    const icons = Array.isArray(icon) ? icon : [icon === true ? undefined : icon];
```
Comment (zombieJ) [inline components/typography/Base.tsx:399]: 这里应该不用判断 `true` 的

File: components/typography/__tests__/index.copy.test.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        iconTexts = [],
+        tooltipTexts = [],
+        tooltipLength,
+      }) {
+        it(name, async () => {
+          jest.useFakeTimers();
+          const wrapper = mount(<Base copyable={{ icon, tooltips }}>test copy</Base>);
+          // icon
+          if (iconClassNames[0] !== undefined) {
+            expect(wrapper.find(iconClassNames[0]).length).toBeTruthy();
```
Comment (zombieJ) [inline components/typography/__tests__/index.copy.test.js:30]: `expect(wrapper.find(iconClassNames[0]).length).toBeTruthy();`
=>
`expect(wrapper.exists(iconClassNames[0])).toBeTruthy();`

File: components/typography/__tests__/index.copy.test.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        tooltipLength: 0,
+      });
+      copyTest({
+        name: 'tooltips first empty 2',
+        tooltips: [''],
+        tooltipLength: 0,
+      });
+    });
+  });
+});
```
Comment (zombieJ) [inline components/typography/__tests__/index.copy.test.js:131]: 貌似 `<><Icon1 /><Icon2 /></>` 应该挂掉的测试没看到？

Comment (crazyair) [reply components/typography/__tests__/index.copy.test.js]:
```
  if (iconClassNames[0] !== undefined) {
     expect(wrapper.exists(iconClassNames[0])).toBeTruthy();
}
```
找 className的话，能找到就能过

Comment (crazyair) [reply components/typography/__tests__/index.copy.test.js]: 增加了 test

File: components/typography/Base.tsx (hunk 2 above)
Comment (crazyair) [reply components/typography/Base.tsx:399]: 为了 icon: true，icons[0] = undefined 现在 copy Icon
```
icons[0] || <CopyOutlined />
```

**Comment by zombieJ** [comment]
> 修复复制支持 <></> 

=>

> 修复 Typography.xxxx 的 xxx 会将 React.Fragment 当做数组处理的问题。

**Comment by afc163** [comment]
图片

**Pull request merged**